File: android/src/main/kotlin/io/stratosfy/stratosfy_scanner/models/Geofence.kt
package io.stratosfy.stratosfy_scanner.models

class Geofence(private val geofence: HashMap<String, Any>) {
    var identifier: String = geofence["identifier"] as String
    var latitude: Double = geofence["latitude"] as Double
    var longitude: Double = geofence["longitude"] as Double
    var radius: Float = (geofence["radius"] as Int).toFloat()

    fun toHashMap(): HashMap<String, Any> {
        return geofence
    }
}

File: lib/models/geofence.dart
part of stratosfy_scanner;

/// Object to represent the geofencing parameter
class Geofence {
  String identifier;
  double latitude, longitude;
  int radius;
  Map<String, dynamic> customData;
  Geofence({
    // Identifier for the geofencing, if you are sending geofencing to a background scanner, identifier is ignored
    required this.identifier,
    // Latitude of the geofence
    required this.latitude,
    // Longitude of the geofence
    required this.longitude,
    // Radius of the geofence, should be in meters and greater than 200
    required this.radius,
    // Custom payload that you want to attach to the geofence. It will be delivered on the native side with enter and exit events
    this.customData = const {},
  }) : assert(radius >= 200);

  Map toMap() {
    return {
      "identifier": identifier,
      "latitude": latitude,
      "longitude": longitude,
      "radius": radius,
      "customData": customData,
    };
  }
}
